include/mesh.h

```diff
@@ -282,6 +290,11 @@ class Mesh {
   //! Return node pairs
   std::vector<std::array<mpm::Index, 2>> node_pairs() const;
 
+  //! Create map of container of particles in sets
+  //! \param[in] map of particles ids in sets
+  void create_particle_sets(
+      std::map<mpm::Index, std::vector<mpm::Index>> particle_sets);
```

Comment:
Pass as const reference
```suggestion
      const std::map<mpm::Index, std::vector<mpm::Index>>& particle_sets);
```

Reply:
could you use `spp::dense_hash_map` here as well?

Reply:
So we need to change the previous PR? Because this input argument coming from IO class. Is that what we need?

Reply:
Yes, sorry! We need to change the output from IO.

Reply:
No problem. Let's do that.

Reply:
Ok, I will do that

Reply:
Sorry I don't find dense_hash_map in "spp", do you mean "spp::sparse_hash_map"?

Reply:
Yes sparse_hash_map